templates/blockonomics_nojs_checkout.php

```diff
@@ -1,70 +1,81 @@
 <?php
+$blockonomics = new Blockonomics;
 $orders = get_option('blockonomics_orders');
-$order_id = $_REQUEST['show_order'];
-$crypto = $_REQUEST['crypto'];
-foreach ($orders[$order_id] as $addr => $order){
-  if ($order['crypto'] == $crypto) break;
-}
-?>
-<div ng-app="shopping-cart-demo">
-  <div ng-controller="CheckoutController">
+$crypto = isset($_REQUEST["crypto"]) ? $_REQUEST["crypto"] : "";
+$order_id = isset($_REQUEST["show_order"]) ? $_REQUEST["show_order"] : "";
+$qrcode = isset($_REQUEST['qrcode']) ? $_REQUEST['qrcode'] : "";
+
+if ($qrcode) {
+  $blockonomics->generate_qrcode($qrcode);
+}else if ($order_id) {
+  $order = $blockonomics->get_order_by_id_and_crypto($orders, $order_id, $crypto);
+  $order_url = $blockonomics->create_order_url($order_id, true);
+  if ($order['status'] >= 0){
+    $finish_order_url = add_query_arg('finish_order', $order['order_id'], $order_url);
+    wp_redirect($finish_order_url);
```

Comment:
This is unecessary, since you are already in php. This is there in app.js coz it is client side. Here you can directly call $blockonomics->redirect_finish_order()

Reply:
Updated in 8496e81